src/coreclr/jit/importercalls.cpp

```diff
@@ -8175,7 +8175,13 @@ void Compiler::impDevirtualizeCall(GenTreeCall*            call,
             return;
         }
 
-        considerGuardedDevirtualization(call, ilOffset, isInterface, baseMethod, baseClass, pContextHandle);
+        CORINFO_CLASS_HANDLE moreExactBaseClass = baseClass;
+        if (objClass != NO_CLASS_HANDLE)
```

Comment:
Don't we already know `objClass` is not null here?

Reply:
> Don't we already know `objClass` is not null here?

Yep, you're correct, removed the condition